packages/kbn-grid-layout/grid/grid_height_smoother.tsx

```diff
@@ -18,61 +18,85 @@ export const GridHeightSmoother = ({
+     * making the panel shrink or grow unpredictably.
+     */
     const interactionStyleSubscription = combineLatest([
       gridLayoutStateManager.gridDimensions$,
       gridLayoutStateManager.interactionEvent$,
     ]).subscribe(([dimensions, interactionEvent]) => {
-      if (!smoothHeightRef.current) return;
-      if (gridLayoutStateManager.expandedPanelId$.getValue()) {
-        return;
-      }
+      if (!smoothHeightRef.current || gridLayoutStateManager.expandedPanelId$.getValue()) return;
+
       if (!interactionEvent) {
         smoothHeightRef.current.style.height = `${dimensions.height}px`;
         smoothHeightRef.current.style.userSelect = 'auto';
         return;
       }
 
-      /**
-       * When the user is interacting with an element, the page can grow, but it cannot
-       * shrink. This is to stop a behaviour where the page would scroll up automatically
-       * making the panel shrink or grow unpredictably.
-       */
       smoothHeightRef.current.style.height = `${Math.max(
         dimensions.height ?? 0,
         smoothHeightRef.current.getBoundingClientRect().height
       )}px`;
       smoothHeightRef.current.style.userSelect = 'none';
     });
 
+    /**
+     * This subscription adds and/or removes the necessary class name for expanded panel styling
+     */
     const expandedPanelSubscription = gridLayoutStateManager.expandedPanelId$.subscribe(
       (expandedPanelId) => {
         if (!smoothHeightRef.current) return;
-
         if (expandedPanelId) {
-          smoothHeightRef.current.style.height = `100%`;
-          smoothHeightRef.current.style.transition = 'none';
+          smoothHeightRef.current.classList.add('kbnGridWrapper--hasExpandedPanel');
         } else {
-          smoothHeightRef.current.style.height = '';
-          smoothHeightRef.current.style.transition = '';
+          smoothHeightRef.current.classList.remove('kbnGridWrapper--hasExpandedPanel');
         }
       }
     );
+
+    /**
+     * This subscription sets global CSS variables that can be used by all components contained within
+     * this wrapper; note that this is **currently** only used for the gutter size, but things like column
+     * count could be added here once we add the ability to change these values
+     */
+    const globalCssVariableSubscription = gridLayoutStateManager.runtimeSettings$
+      .pipe(
+        map(({ gutterSize }) => gutterSize),
+        distinctUntilChanged()
+      )
+      .subscribe((gutterSize) => {
+        smoothHeightRef.current?.style.setProperty('--kbnGridGutterSize', `${gutterSize}`);
+      });
+
     return () => {
       interactionStyleSubscription.unsubscribe();
       expandedPanelSubscription.unsubscribe();
+      globalCssVariableSubscription.unsubscribe();
     };
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   return (
     <div
       ref={smoothHeightRef}
+      className={'kbnGridWrapper'}
```

Comment:
if you decide to follow my comment from https://github.com/elastic/kibana/pull/205341/files#r1910325248 I think this class is unneccessary
```suggestion
```

Reply:
See https://github.com/elastic/kibana/pull/205341#discussion_r1910605099